src/Bounds.cpp

```diff
@@ -37,6 +37,45 @@ int static_sign(Expr x) {
     }
     return 0;
 }
+
+
+// Try to find a constant that is:
+// greater than or equal to the given Expr (if d is Up), or
+// less than or equal to the given Expr (if d is Down).
+enum Direction {Up, Down};
```

Comment:
I'd sort of prefer this enum were "Upper" and "Lower" and is "upper bound" (constant greater than or equal to any value) and "lower bound" (constant that is less than or equal to any value).